extensions/transform/org.eclipse.smarthome.transform.regex/README.md

```diff
@@ -1,18 +1,87 @@
 # RegEx Transformation Service
 
-Given a source string and a regular expression, use the regular expression to yield a transformed string.  
+Transforms a source string on basis of the regular expression (regex) search pattern to a defined result string.
 
-If the regular expression is in the format `s/<regex>/result/g`, replace all occurrences of `<regex>` in the source string with `result` and return the result.
+The simplest regex is in the form `<regex>` and transforms the input string on basis of the regex patern to a result string.
+A full regex is in the form `s/<regex>/<result>/g` whereat the delimiter `s` and the regex flag `g` have special meaning.
 
-If the regular expression is in the format `s/<regex>/result/`, replace the first occurrence of `<regex>` in the source string with `result` and return the result.
+The regular expression in the format `s/<regex>/result/g`, replaces all occurrences of `<regex>` in the source string with `result`.
+The regular expression in the format `s/<regex>/result/` (without `g`), replaces the first occurrence of `<regex>` in the source string with `result`.
 
-If the regular expression contains a [capture group](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#cg), return the captured string.  The regular expression in this case is further restricted to isolate a complete line by adding `^` to the beginning and `$` to the end.
+If the regular expression contains a [capture group](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#cg), it returns the captured string.
+The regular expression in this case is further restricted to isolate a complete line by adding `^` to the beginning and `$` to the end.
+
+The special characters `\.[]{}()*+-?^$|` have to be escaped when they should be used as literal character.
 
 ## Examples
 
-With the input string `My network does not work.`:
+### Basic Examples
+
+|         Input String        |    Regular Expression    |         Output String        | Explanation              |
+|---------------------------|------------------------|----------------------------|--------------------------|
+| `My network does not work.` | `s/work/cast/g` | `"My netcastdoes not cast."` | Replaces all matches with regex |
```

Comment:
"My netcastdoes not cast."` -> "My netcast does not cast."`
because the space will not be removed, right?

"Replaces all matches with regex" -> "Replaces all matches of the string "work" with the string "cast""